### DISCUSSION TITLE
Floating elements inside shadow doms incorrectly positioned in Firefox

### DISCUSSION THREAD

**Discussion by lennartbuit** [floating-ui/floating-ui#1900] [Q&A]
Hey, I would like to file a bug, but I'm not particularly sure what the issue is.

I have custom elements ("webcomponents") that render a floating element, and this floating element might contain another floating element. For example, a popup that contains a select with a little menu.

This works in chrome, the "inner" floating element is nicely positioned relative to the reference in the "outer" floating element. In Firefox, however, it appears to add the distances from the body's origin to the floating element.

Visually this is how it looks in chrome, this is what I expect:

<img width="125" alt="Screenshot 2022-09-23 at 15 28 46" src="https://user-images.githubusercontent.com/37534243/191971067-0a37ba60-76a1-44ad-bf15-cafe557365f7.png">

So, Abc is a reference for a floating element that contains Ghi, Ghi is the reference for Jkl.

In Firefox, however, it looks like this:

<img width="169" alt="Screenshot 2022-09-23 at 15 29 49" src="https://user-images.githubusercontent.com/37534243/191971198-86ab2a54-c73d-4abb-8f2b-239b89101043.png">

Here is it in sandbox form: https://codesandbox.io/s/headless-leftpad-m34s33?file=/src/index.js

Now; I've seen some chatter on this topic related to shadow roots, but I must admit that I don't fully understand their conclusions. Could you help me understand why it behaves like this in Firefox and what to do about it?

**Comment by atomiks** [comment]
You're hitting this issue: https://github.com/floating-ui/floating-ui/pull/1894 (works in 1.0.1, reverted in 1.0.2). It's also broken in Safari and will probably break in Chrome in the future soon. It had to be reverted because it was causing poor performance as a default.

Best option is to avoid the DOM layout you have somehow, or if the performance cost is acceptable in some scenarios, copy the polyfill that was deleted in that file like:

```js
import {platform, computePosition} from '@floating-ui/dom';

computePosition(..., {
  platform: {
    ...platform,
    getOffsetParent: () => withThePolyfill,
  }
});
```

**Reply by lennartbuit** [reply]
(Reposted because I did not know GH discussions was threaded)

I guess I'm missing something. I checked with 1.0.1, and that indeed does work. I'm currently trying to (optionally) allow the use of the polyfill, but that requires me to inline the entirety of `packages/dom/src/utils/getOffsetParent.ts` into my codebase _with_ all its dependencies. Basically making like a copy of the utils dir of floating-ui. Is that the intention?

**Reply by atomiks** [reply]
Unfortunately yeah, there's no real other way around it. It causes poor performance in various scenarios so it's really not great to begin with.
